### PR TITLE
Create helm chart

### PR DESCRIPTION
**Pull request by muralov**
**Description**
Convert Kustomize resources to Helm chart.

**Related issue(s)**
#1

### CODE REVIEW

File: config/external/crds/operator.kyma-project.io_kymas.yaml
Comment (mfaizanse) [inline config/external/crds/operator.kyma-project.io_kymas.yaml:1]: (nit) Can you add a comment on top of this file that it is only used for testing purposes?

File: config/helm/eventing-auth-manager/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        control-plane: controller-manager
+      {{- include "eventing-auth-manager.selectorLabels" . | nindent 8 }}
+      annotations:
+        kubectl.kubernetes.io/default-container: manager
+    spec:
+      containers:
+      - args:
+        - --health-probe-bind-address=:8081
+        - --metrics-bind-address=127.0.0.1:8080
+        - --leader-elect
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/templates/deployment.yaml:43]: What the purpose of `--leader-elect` arg?

File: config/helm/eventing-auth-manager/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        command:
+        - /manager
+        image: {{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag
+          | default .Chart.AppVersion }}
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/templates/deployment.yaml:47]:
```
image: {{ .Values.manager.image.repository }}:{{ .Values.manager.image.tag }}
```

File: config/helm/eventing-auth-manager/templates/leader-election-rbac.yaml [hunk 3]
```diff
@@ -0,0 +1,61 @@
+apiVersion: rbac.authorization.k8s.io/v1
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/templates/leader-election-rbac.yaml:1]: Do we have leader elections? If not, then we do not need RBAC for them?

File: config/helm/eventing-auth-manager/templates/metrics-reader-rbac.yaml [hunk 4]
```diff
@@ -0,0 +1,14 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/templates/metrics-reader-rbac.yaml:2]: Why do we need this ClusterRole?

File: config/helm/eventing-auth-manager/templates/metrics-service.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
+apiVersion: v1
+kind: Service
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/templates/metrics-service.yaml:2]: Why do we need a service for EAM? Does some external components sends a REST call to EAM?

File: config/helm/eventing-auth-manager/values.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
+manager:
+  image:
+    repository: docker.io/muralov/eventing-auth-manager
```
Comment (mfaizanse) [inline config/helm/eventing-auth-manager/values.yaml:3]: Do we have a prow image?

File: config/helm/eventing-auth-manager/templates/deployment.yaml (hunk 1 above)
Comment (muralov) [reply config/helm/eventing-auth-manager/templates/deployment.yaml:43]: this argument used by the `ctl.Manager` for leader-election [here](https://github.com/muralov/eventing-auth-manager/blob/24168790d55ce225802480a9b5de4ac44aab8c4a/cmd/main.go#L65)

File: config/helm/eventing-auth-manager/templates/metrics-service.yaml (hunk 5 above)
Comment (muralov) [reply config/helm/eventing-auth-manager/templates/metrics-service.yaml:2]: This is just metrics service to expose metrics.

File: config/helm/eventing-auth-manager/templates/metrics-reader-rbac.yaml (hunk 4 above)
Comment (muralov) [reply config/helm/eventing-auth-manager/templates/metrics-reader-rbac.yaml:2]: Good question! However, it's not worked with just Role/RoleBinding. I think, it needs ClusterRole to retrieve the resources from the cache.

File: config/helm/eventing-auth-manager/templates/leader-election-rbac.yaml (hunk 3 above)
Comment (mfaizanse) [reply config/helm/eventing-auth-manager/templates/leader-election-rbac.yaml:1]: I mean will we use leader election? Do we have >1 replicas of EAM?

File: config/helm/eventing-auth-manager/templates/metrics-service.yaml (hunk 5 above)
Comment (mfaizanse) [reply config/helm/eventing-auth-manager/templates/metrics-service.yaml:2]: Who will use this service?

**Comment by mfaizanse** [comment]
Please keep the kustomize files and add a helm chart rendered from it.

**Pull request closed without merging**